app/src/main/java/com/x8bit/bitwarden/ui/auth/feature/vaultunlock/VaultUnlockViewModel.kt

```diff
@@ -138,9 +139,17 @@ class VaultUnlockViewModel @Inject constructor(
             is VaultUnlockAction.BiometricsUnlockSuccess -> handleBiometricsUnlockSuccess(action)
             VaultUnlockAction.UnlockClick -> handleUnlockClick()
             is VaultUnlockAction.Internal -> handleInternalAction(action)
+            VaultUnlockAction.BiometricsNoLongerSupported -> handleBiometricsNoLongerSupported()
         }
     }
 
+    private fun handleBiometricsNoLongerSupported() {
+        specialCircumstanceManager.specialCircumstance =
+            SpecialCircumstance.PreAuthEvent.BiometricsNoLongerSupported
```

Comment:
Maybe this has been discussed already but instead of dealing of this special circumstance, could we show the dialog here and log the user out when they click OK?

Reply:
yeah I think you are 100% could just set pending addition here by itself